validator/client/wait_for_activation.go

```diff
@@ -129,3 +148,24 @@ func incrementRetries(ctx context.Context) context.Context {
 	attempts := streamAttempts(ctx)
 	return context.WithValue(ctx, waitForActivationAttemptsContextKey, attempts+1)
 }
+
+func handleAccountsChanged(ctx context.Context, v Validator, accountsChangedChan chan<- struct{}) {
```

Comment:
This method seems redundant, its basically creates a channel to relay account changes to the `accountsChangedChan` . How about changing the `accountsChangedChan` type to a ` chan<- [][48]byte` and then subscribe account changes to it directly ? You could skip this additional routine and channel being created here.

Reply:
Excellent feedback! Doing this now.